pkg/controller/stacks/install/stackinstall_test.go

```diff
@@ -622,3 +631,225 @@ func TestHandlerFactory(t *testing.T) {
+		},
+		{
+			name: "StillInUseMultiParentLabels",
+			fields: fields{
+				ext: resource(),
+				clientFunc: func() client.Client {
+					c := crd(withCRDGroupKind(group, kind), withCRDVersion(version), withCRDLabels(map[string]string{stacks.LabelKubernetesManagedBy: "stack-manager", label: "true"}))
+					return fake.NewFakeClient(&c)
+				},
+			},
+			want: []apiextensions.CustomResourceDefinition{crd(withCRDGroupKind(group, kind), withCRDVersion(version), withCRDLabels(map[string]string{stacks.LabelKubernetesManagedBy: "stack-manager", label: "true"}))},
+		},
+		{
+			name: "SafeToDelete",
+			fields: fields{
+				ext: resource(),
+				clientFunc: func() client.Client {
+					c := crd(withCRDGroupKind(group, kind), withCRDVersion(version), withCRDLabels(map[string]string{stacks.LabelKubernetesManagedBy: "stack-manager"}))
+					return fake.NewFakeClient(&c)
+				},
+			},
+			unwanted: []apiextensions.CustomResourceDefinition{crd(withCRDGroupKind(group, kind))},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+
+			h := &stackInstallHandler{
+				kube: tt.fields.clientFunc(),
+				ext:  tt.fields.ext,
+				log:  logging.NewNopLogger(),
+			}
+			gotErr := h.deleteOrphanedCRDs(context.TODO())
+
+			if diff := cmp.Diff(tt.wantErr, gotErr, test.EquateErrors()); diff != "" {
+				t.Fatalf("stackHandler.deleteOrphanedCRDs(...): -want error, +got error: %s", diff)
+			}
+
+			if tt.want != nil {
+				for _, wanted := range tt.want {
+					got := &apiextensions.CustomResourceDefinition{}
+					assertKubernetesObject(t, g, got, &wanted, h.kube)
+				}
+			}
+
+			if tt.unwanted != nil {
+				for _, unwanted := range tt.unwanted {
+					got := &apiextensions.CustomResourceDefinition{}
+					assertNoKubernetesObject(t, g, got, &unwanted, h.kube)
+				}
+			}
+
+		})
+	}
+}
+
+type objectWithGVK interface {
+	runtime.Object
+	metav1.Object
+}
+
+func assertKubernetesObject(t *testing.T, g *GomegaWithT, got objectWithGVK, want metav1.Object, kube client.Client) {
+	n := types.NamespacedName{Name: want.GetName(), Namespace: want.GetNamespace()}
+	g.Expect(kube.Get(ctx, n, got)).NotTo(HaveOccurred())
+
+	// NOTE(muvaf): retrieved objects have TypeMeta and
+	// ObjectMeta.ResourceVersion filled but since we work on strong-typed
+	// objects, we don't need to check them.
+	got.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{})
+	got.SetResourceVersion(want.GetResourceVersion())
+
+	if diff := cmp.Diff(want, got, test.EquateConditions()); diff != "" {
+		t.Errorf("-want, +got:\n%s", diff)
+	}
+}
+
+func assertNoKubernetesObject(t *testing.T, g *GomegaWithT, got runtime.Object, unwanted metav1.Object, kube client.Client) {
+	n := types.NamespacedName{Name: unwanted.GetName(), Namespace: unwanted.GetNamespace()}
+	g.Expect(kube.Get(ctx, n, got)).To(HaveOccurred())
```

Comment:
I took a stab at this, but within this function I can't replicate the parameters needed to match `kerrors.NewNotFound()`.  And after looking into how to do this with Gometa, I started wondering why we are introducing Gomega at all.  This `assertNoKubernetesObject` mirrors `assertKubernetesObject` which were already present in either stackinstall_test or stack_test (I don't recall which came first).

In a future PR I think we can remove this because I don't think we need the Gomega semantics and detailed reported over what is already available and used elsewhere with `Diff`.

Reply:
I am always in support of removing Gomega usage.